lib/Cake/Test/Case/Controller/Component/CookieComponentTest.php

```diff
@@ -790,9 +752,99 @@ public function testDeleteChildrenNotExist() {
 	}
 
 /**
+ * Test that deleting a child entry of the cookie array works
+ *
+ * @return void
+ */
+	public function testDeleteChildEntry() {
+		$response = $this->Controller->response;
+
+		$this->Cookie->write('User.name', 'user', false);
+		$this->Cookie->write('User.email', 'user@example.com', false);
```

Comment:
Sorry for my lack of words. I am guessing that you are thinking about sending a single serialized cookie. But I was thinking about sending Set-Cookie header for each individual entry by using `Hash::flatten()`:
```php
foreach (Hash::flatten($cookies) as $name => $value) {
    $name = 'CakeCookie[' . str_replace('.', '][', $name) . ']';
    $response->cookie(compact('name', 'value'));
}
```

To be clear, let's assume the concerned example in #2182.
```php
$this->Cookie->write('User',
    array('name' => 'Larry', 'role' => 'Lead')
);
echo $this->Cookie->read('User.name'); //'Larry'

$this->Cookie->write('User.name', 'coco'); 
echo $this->Cookie->read('User.name'); //'coco'

$this->Cookie->delete('User.name'); 
echo $this->Cookie->read('User.name'); //'Larry'  !!!
```

Currently, the Set-Cookie headers will be like the following:
```
Set-Cookie: CakeCookie[User]={"name":"Larry","role":"Lead"}; path=/
Set-Cookie: CakeCookie[User][name]=coco; path=/
Set-Cookie: CakeCookie[User][name]=deleted; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/
```
Since the expired cookie destroys only `CakeCookie[User][name]`, the `Larry` in the cookie named `CakeCookie[User]` will be back.

But if we send Set-Cookie header for each individual entry, the result will be:
```
Set-Cookie: CakeCookie[User][name]=Larry; path=/
Set-Cookie: CakeCookie[User][role]=Lead; path=/
Set-Cookie: CakeCookie[User][name]=coco; path=/
Set-Cookie: CakeCookie[User][name]=deleted; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/
```
In this case, the expired cookie will destroy the cookie named `CakeCookie[User][name]` completely.